extension/test/test-logger.js

```diff
@@ -10,7 +10,13 @@ var verifyAsyncOutput = function (test, logger, testFunc) {
   var message = '';
 
   return function (status) {
-    var line = {}, hasMore;
+    var line = {}, hasMore, isSuccessStatus;
+
+    isSuccessStatus = typeof status == "number";
+    test.assert(isSuccessStatus, status["message"] ? status.message : status);
+    if (!isSuccessStatus) {
+      return;
+    }
```

Comment:
I still have problems to understand this code. Isn't `test.assert` an instance of the assertion class? Shouldn't you use a specific test method here? It would be also good to add at least a one-liner for documentation what those lines are doing.